casper/src/main/scala/coop/rchain/casper/util/comm/DumpProgram.scala

```diff
@@ -0,0 +1,49 @@
+package coop.rchain.casper.util.comm
+import java.io.BufferedWriter
+import java.nio.charset.Charset
+import java.nio.file.{Files, Paths}
+import java.time.format.DateTimeFormatter
+
+import cats.effect.{Resource, Sync}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.apply._
+import coop.rchain.shared.DateTime
+
+import scala.language.higherKinds
+
+class DumpProgram[F[_]: Sync: DeployService: DateTime] {
+  val formatter      = DateTimeFormatter.ofPattern("YYYY-mm-dd_HH-mm-ss")
+  val charset        = Charset.forName("UTF-8")
+  val standardPrefix = "node_dump"
+
+  private def filename: F[String] =
+    for {
+      dateTime  <- DateTime[F].dateTime
```

Comment:
Do we really need all that code to do
```scala
dateTime <- Sync[F].delay(LocalDateTime.now())
```
?

In my opinion this is an overkill.

Reply:
Not sure about that btw, open for discussion.
My impression was that it might be helpful to have such functionality, but it doesn't make sense if we will never use it again ofc.